tests/framework/clients/rancher/provisioning/cluster.go

```diff
@@ -0,0 +1,97 @@
+package provisioning
+
+import (
+	"context"
+
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewRKE2ClusterConfig(clusterName, namespace, cni, cloudCredentialSecretName, kubernetesVersion string, machinePools []apisV1.RKEMachinePool) *apisV1.Cluster {
+	typeMeta := metav1.TypeMeta{
+		Kind:       "Cluster",
+		APIVersion: "provisioning.cattle.io/v1",
+	}
+
+	//metav1.ObjectMeta
+	objectMeta := metav1.ObjectMeta{
+		Name:      clusterName,
+		Namespace: namespace,
+	}
+
+	etcd := &rkev1.ETCD{
+		DisableSnapshots:     false,
+		S3:                   nil,
+		SnapshotRetention:    5,
+		SnapshotScheduleCron: "0 */5 * * *",
+	}
+
+	chartValuesMap := rkev1.GenericMap{
+		Data: map[string]interface{}{},
+	}
+
+	machineGlobalConfigMap := rkev1.GenericMap{
+		Data: map[string]interface{}{
+			"cni":                 cni,
+			"disable-kube-proxy":  false,
+			"etcd-expose-metrics": false,
+			"profile":             nil,
+		},
+	}
+
+	localClusterAuthEndpoint := rkev1.LocalClusterAuthEndpoint{
+		CACerts: "",
+		Enabled: false,
+		FQDN:    "",
+	}
+
+	upgradeStrategy := rkev1.ClusterUpgradeStrategy{
+		ControlPlaneConcurrency:  "10%",
+		ControlPlaneDrainOptions: rkev1.DrainOptions{},
+		WorkerConcurrency:        "10%",
+		WorkerDrainOptions:       rkev1.DrainOptions{},
+	}
+
+	rkeSpecCommon := rkev1.RKEClusterSpecCommon{
+		ChartValues:           chartValuesMap,
+		MachineGlobalConfig:   machineGlobalConfigMap,
+		ETCD:                  etcd,
+		UpgradeStrategy:       upgradeStrategy,
+		MachineSelectorConfig: []rkev1.RKESystemConfig{},
+	}
+
+	rkeConfig := &apisV1.RKEConfig{
+		RKEClusterSpecCommon: rkeSpecCommon,
+		MachinePools:         machinePools,
+	}
+
+	spec := apisV1.ClusterSpec{
+		CloudCredentialSecretName: cloudCredentialSecretName,
+		KubernetesVersion:         kubernetesVersion,
+		LocalClusterAuthEndpoint:  localClusterAuthEndpoint,
+
+		RKEConfig: rkeConfig,
+	}
+
+	v1Cluster := &apisV1.Cluster{
+		TypeMeta:   typeMeta,
+		ObjectMeta: objectMeta,
+		Spec:       spec,
+	}
+
+	return v1Cluster
+}
+
+func (c *Cluster) Create(ctx context.Context, cluster *apisV1.Cluster, opts metav1.CreateOptions) (*apisV1.Cluster, error) {
```

Comment:
isn't this already handled with the api base client?

Reply:
No since this is a v1 client its handled differently than the other v3 clients. ProvisioningClient is an interface, so I did it this way